File: ruby/grep.rb
def grep(term, file)
	File.foreach(file) { |line|
        puts line if line.include? term
    }
end

if ( ARGV.length < 2 )
    puts "usage: grep term file"
else
	term = ARGV[0]
    file = ARGV[1]
    grep(ARGV[0], file)
end

File: python/grep.py
import sys

def grep(term, file):
	with open(file) as file:
	    for line in file:
	        if term in line: 
	   			print line		

if len(sys.argv) < 3 :
	print "usage: grep term file"
else: 
	term = sys.argv[1]
	file = sys.argv[2]
	grep(term, file)